clang/lib/AST/Stmt.cpp

```diff
@@ -609,9 +632,10 @@ int GCCAsmStmt::getNamedOperand(StringRef SymbolicName) const {
 /// true, otherwise return false.
 unsigned GCCAsmStmt::AnalyzeAsmString(SmallVectorImpl<AsmStringPiece>&Pieces,
                                 const ASTContext &C, unsigned &DiagOffs) const {
-  StringRef Str = getAsmString()->getString();
-  const char *StrStart = Str.begin();
-  const char *StrEnd = Str.end();
+
+  std::string Str = getAsmString();
+  const char *StrStart = Str.data();
```

Comment:
Curious why data/data+size instead of just begin/end?

Reply:
begin/end are iterators, and some of the arithmetic in this function we do requires pointers

Reply:
.... Isn't the type of the iterators `const char*`?